java/src/main/java/ai/rapids/cudf/MemoryCleaner.java

```diff
@@ -144,11 +143,35 @@ public void clean() {
     }
   }
 
+  /**
+   * The default GPU as set by user threads.
+   */
+  private static volatile int defaultGpu = -1;
+  /**
+   * The default GPU as set by the cleaner thread.
+   */
+  private static volatile int currentGpuId = -1;
```

Comment:
I'm not sure why this is static and volatile, as it appears to only be accessed within the cleaner thread.   Can this just be  a local variable within the thread's `run()` method?